src/app/qgisapp.cpp

```diff
@@ -5480,22 +5484,37 @@ QgsMeshLayer *QgisApp::addMeshLayerPrivate( const QString &url, const QString &b
     return nullptr;
   }
 
-  // Manage default reference time, if not reference time is present by default in the layer -> assign one
-  if ( ! layer->temporalProperties()->referenceTime().isValid() )
+  if ( !layer->isValid() && layer->subLayers().count() > 0 ) //sublayers to load
   {
-    QDateTime referenceTime = QgsProject::instance()->timeSettings()->temporalRange().begin();
-    if ( !referenceTime.isValid() ) // If project reference time is invalid, use current date
-      referenceTime = QDateTime( QDate::currentDate(), QTime( 0, 0, 0, Qt::UTC ) );
-    layer->temporalProperties()->setReferenceTime( referenceTime, layer->dataProvider()->temporalCapabilities() );
-  }
+    QList< QgsMapLayer * > subLayers = askUserForMDALSublayers( layer.get() );
 
-  QgsProject::instance()->addMapLayer( layer.get() );
-
-  askUserForDatumTransform( layer->crs(), QgsProject::instance()->crs(), layer.get() );
+    if ( !subLayers.isEmpty() )
+    {
+      for ( QgsMapLayer *newLayer : qgis::as_const( subLayers ) )
+      {
+        askUserForDatumTransform( newLayer->crs(), QgsProject::instance()->crs(), newLayer );
+        // Manage default reference time, if not reference time is present by default in the layer -> assign one
+        if ( ! layer->temporalProperties()->referenceTime().isValid() )
+          layer->temporalProperties()->setReferenceTime( referenceTime, layer->dataProvider()->temporalCapabilities() );
+        bool ok;
+        newLayer->loadDefaultStyle( ok );
+        newLayer->loadDefaultMetadata( ok );
+      }
+    }
+    layer.reset( !subLayers.isEmpty() ? qobject_cast< QgsMeshLayer * >( subLayers.at( 0 ) ) : nullptr );
```

Comment:
you do not need to test subLayers.isEmpty(), you are already in the if branch for this condition

Reply:
we are outside the if condition, but I could go in and add a else...